### PR TITLE
upgrade filter library to version 1.2

### PR DESCRIPTION
**Pull request by tyrasd**
which includes the following changes: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/libs/ohsome-filter/-/blob/1.2.0/CHANGELOG.md#anchor-120

### CODE REVIEW

**Review by FabiKo117 — CHANGES_REQUESTED** [review]
Please also add these new features in the docs of the ohsome API under the dedicated [filter page](https://docs.ohsome.org/ohsome-api/v1/filter.html).

File: docs/filter.rst [hunk 1]
```diff
@@ -124,18 +137,16 @@ Here's some useful examples for querying some OSM features:
     |                  |                                                        | | source), but is in reality |
     |                  |                                                        | | not one.                   |
     +------------------+--------------------------------------------------------+------------------------------+
-    | | highways       | | ``type:way and (highway=motorway or``                | | The list of used tags      |
```
Comment (tyrasd) [inline docs/filter.rst:127]: this old clumsy example can now be written much more nicely with the new `key in (value list)` syntax :sunglasses:

Comment (FabiKo117) [reply docs/filter.rst:127]: true, thanks for updating that 👍

**Review by FabiKo117 — APPROVED** [review]
looks good now 👍

**Pull request merged**